specs/data_structures.md

```diff
@@ -161,11 +171,9 @@ https://developers.google.com/protocol-buffers/docs/proto3
 
 # Hashing
 
-https://en.wikipedia.org/wiki/SHA-3
+All protocol-level hashing is done using [Keccak-256](https://keccak.team/keccak.html), and not SHA3-256 ([FIPS 202](https://keccak.team/specifications.html#FIPS_202)). This is to enable compatibility with [Ethereum](https://ethereum.org)'s EVM. Keccak-256 outputs a digest that is 256 bits (i.e. 32 bytes) long.
 
-https://godoc.org/golang.org/x/crypto/sha3
-
-https://docs.rs/sha3
+Libraries implementing Keccak-256 are available in Go (https://godoc.org/golang.org/x/crypto/sha3) and Rust (https://docs.rs/sha3).
 
 # Signing
```

Comment:
Note: for signing consensus data (votes etc), the structs are still binary/proto3 encoded but "canonicalized" and length prefixed:
https://github.com/tendermint/tendermint/blob/aeb6cc475e14c9b4c4455111f6dfb54672507006/types/vote.go#L83-L89

Reply:
It used to be JSON data but apparently this was too expensive to parse in the evm: https://github.com/tendermint/tendermint/issues/1622